src/chunk.c

```diff
@@ -21,6 +21,25 @@ void Uncompressed_FreeChunk(Chunk_t *chunk) {
     free(chunk);
 }
 
+Chunk_t *Uncompressed_SplitChunk(Chunk_t *chunk) {
+    Chunk *curChunk = (Chunk *)chunk;
+    size_t split = curChunk->num_samples / 2;
+    size_t curNumSamples = curChunk->num_samples - split;
```

Comment:
split and curNumSamples are always equal, we can just use split.

Reply:
They don't have to be equal. Either we deleted one before or in uncompressed it can be anything.

Reply:
@ashtul maybe im missing something, both `curNumSamples` and `split` do not change post initlization.
So if:
split = n /2
cur = n - split -> n - n/2 -> (2n - n)/2 -> n/2

Reply:
We are using integers
7 / 2 = 3
7 - 3 = 4